lib/private/L10N/Factory.php

```diff
@@ -58,6 +58,10 @@ class Factory implements IFactory {
 		'pt_BR', 'pt_PT', 'da', 'fi_FI', 'nb_NO', 'sv', 'tr', 'zh_CN', 'ko'
 	];
 
+	public const RTL_LANGUAGES = [
+		'ar', 'dv', 'fa', 'ha', 'he', 'ks', 'ku', 'ps', 'sd', 'ug', 'ur_PK', 'uz', 'yi',
```

Comment:
Not sure why you removed this @nickvergessen ?
https://www.localeplanet.com/icu/ur-IN/index.html
https://www.localeplanet.com/icu/ur/index.html
```suggestion
		'ar', 'dv', 'fa', 'ha', 'he', 'ks', 'ku', 'ps', 'sd', 'ug', 'ur', 'ur_IN', 'ur_PK', 'uz', 'yi',
```

Reply:
Because we use 2 character [ISO 639-1](https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes) for translations. I removed everything that is not available at all on transifex.

I see now that despite the `RTL_LANGUAGE` name of the const it seems to be taking the orientation based on the locale, and I'm not sure about that either.

Reply:
I will try to get input from couple of RTL speakers to see if that is intended and also check what other projects do.
Then we either rename the constant to locale and revert the list, or we recheck my list.

Reply:
JS lib seems to do something wrong as well:
https://github.com/nextcloud-libraries/nextcloud-l10n/blob/e0b16b54fc946273e31e99e0b16d5e85b4ff9560/lib/locale.ts#L28

Takes the language (which is always 2 letters or `ab_YZ` and compares it to 3 letter locales

Reply:
But let's wait with adjusting until we know

Reply:
@nickvergessen the direction (RTL vs LTR) is get by the locale not the language.
So we need to consider all valid locales and we also support 3 digit locales. Our local validation considers all locales from `resources/locales.json` as valid -> all the locales I mentioned are included there.

Reply:
> the direction (RTL vs LTR) is get by the locale not the language.

If this is the study fact (and not just describing what is done in this PR here), we shall rename the constant, revert the list and fix the JS libary as it is using the language and not the locale to detect RTL.

Reply:
This makes sense (but also its sad we use 639-1 instead of 639-2)

Reply:
I had a look at our transifex page, there seem to be some `639-2` languages, not that popular but with ~1% and growing translations. Do we just disable merging it into server?

Reply:
Current feedback is that both would expect it based on language as that defines the direction of most text.